### PR TITLE
Support added for SYSLOG

### PR DESCRIPTION
**Pull request by swarudka**
- added support for SYSLOG in existing configuration

### CODE REVIEW

File: gen/definitions/router_syslog_vrf.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+---
+name: syslog_vrf
+path: Cisco-IOS-XR-um-logging-cfg:/logging/vrfs/vrf[vrf-name=%s]
+doc_category: logging
+attributes:
+  - yang_name: vrf-name
+    id: true
+    example: CORE-Mgmt
+
+  - yang_name: host-ipv4-addresses/host-ipv4-address
```
Comment (danischm) [inline gen/definitions/router_syslog_vrf.yaml:10]: When we add this for 'host-ipv4-addresses', let's add the same for ipv6 and names as well.

File: go.mod [hunk 2]
```diff
@@ -159,7 +159,7 @@ require (
 	go.opencensus.io v0.24.0 // indirect
 	go.uber.org/atomic v1.10.0 // indirect
 	go4.org/intern v0.0.0-20220617035311-6925f38cc365 // indirect
-	go4.org/unsafe/assume-no-moving-gc v0.0.0-20220617031537-928513b29760 // indirect
+	go4.org/unsafe/assume-no-moving-gc v0.0.0-20230221090011-e4bae7ad2296 // indirect
```
Comment (danischm) [inline go.mod:162]: Is there a specific reason you updated this dependency?

**Review by danischm — CHANGES_REQUESTED** [review]

File: gen/definitions/router_syslog.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
+---
```
[the remark this answers is not in the document]
Comment (swarudka) [reply gen/definitions/router_syslog.yaml]: Corrected.

File: gen/definitions/router_syslog.yaml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+name: syslog
+path: Cisco-IOS-XR-um-logging-cfg:/logging
+doc_category: logging
```
[the remark this answers is not in the document]
Comment (swarudka) [reply gen/definitions/router_syslog.yaml]: Added the category as Management.

File: gen/definitions/router_syslog.yaml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+attributes:
+  - yang_name: ipv4/dscp
+    example: cs6
+  - yang_name: trap
+    type: true
```
[the remark this answers is not in the document]
Comment (swarudka) [reply gen/definitions/router_syslog.yaml]: It was added just for my reference, Now its corrected.

File: gen/definitions/router_syslog.yaml [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+  - yang_name : buffered/level                  
+    example : debugging
+  - yang_name : facility/level
+    example : local7
+  - yang_name : hostnameprefix
+    example : ALMTX1P01
+  - yang_name : suppress/duplicates
+    example : true
+  - yang_name : source-interfaces/source-interface
+    tf_name : logging_source_interfaces
```
[the remark this answers is not in the document]
Comment (swarudka) [reply gen/definitions/router_syslog.yaml]: Name updated as per suggested  "source_interfaceses"

File: gen/definitions/router_syslog_vrf.yaml (hunk 1 above)
Comment (swarudka) [reply gen/definitions/router_syslog_vrf.yaml]: support added for both ipv4 & ipv6

File: go.mod (hunk 2 above)
Comment (swarudka) [reply go.mod:162]: I was facing issue initially while running go generate.
Error pic attached.
As per the error i have added this dependency & able to run go generate

image

**Review by danischm — CHANGES_REQUESTED** [review]

File: gen/definitions/logging_vrf.yaml [hunk 7]
```diff
@@ -0,0 +1,27 @@
+---
+name: Logging_vrf
```
Comment (danischm) [inline gen/definitions/logging_vrf.yaml:2]: This one should be updated to `Logging VRF`.

File: examples/data-sources/iosxr_syslog/data-source.tf [hunk 8]
```diff
@@ -0,0 +1,2 @@
+data "iosxr_syslog" "example" {
```
Comment (danischm) [inline examples/data-sources/iosxr_syslog/data-source.tf:1]: This file needs to be deleted manually.

File: examples/data-sources/iosxr_syslog_vrf/data-source.tf [hunk 9]
```diff
@@ -0,0 +1,3 @@
+data "iosxr_syslog_vrf" "example" {
```
Comment (danischm) [inline examples/data-sources/iosxr_syslog_vrf/data-source.tf:1]: This file needs to be deleted manually.

**Review by danischm — CHANGES_REQUESTED** [review]

File: gen/definitions/logging.yaml [hunk 10]
```diff
@@ -0,0 +1,35 @@
+---
+name: Logging
+path: Cisco-IOS-XR-um-logging-cfg:/logging
```
Comment (danischm) [inline gen/definitions/logging.yaml:3]: Small cosmetic thing, we can remove the first '/' from the path and change this to: `Cisco-IOS-XR-um-logging-cfg:logging`.

File: gen/definitions/logging_vrf.yaml [hunk 11]
```diff
@@ -0,0 +1,27 @@
+---
+name: Logging VRF
+path: Cisco-IOS-XR-um-logging-cfg:/logging/vrfs/vrf[vrf-name=%s]
```
Comment (danischm) [inline gen/definitions/logging_vrf.yaml:3]: same as above

**Review by danischm — CHANGES_REQUESTED** [review]

**Review by danischm — APPROVED** [review]

**Pull request merged**